w3af/plugins/audit/xss.py

```diff
@@ -134,18 +146,38 @@ def _report_vuln(self, mutant, response, mod_value):
         
         self.kb_append_uniq(self, 'xss', v)
 
-    def _identify_trivial_xss(self, mutant):
+    def _get_context_payloads(self, mutant):
+        """
+        Returns payload for each of the context where mutant was reflected
+
+        :return: A set of payloads
+        """
+        payload = place_boundary('')
+
+        context_mutant = mutant.copy()
+        context_mutant.set_token_value(payload)
+
+        response = self._uri_opener.send_mutant(context_mutant)
+        body = response.get_body()
+
+        payloads = set()
+        for context in get_context_iter(body, get_boundary(payload)):
+            payloads.update(context.get_payloads())
+
+        return payloads
+
+    def _identify_trivial_xss(self, mutant, payloads):
         """
         Identify trivial cases of XSS where all chars are echoed back and no
         filter and/or encoding is in place.
         
         :return: True in the case where a trivial XSS was identified.
         """
-        payload = replace_randomize(''.join(self.PAYLOADS))
+        payload = place_boundary(''.join(payloads))
 
         trivial_mutant = mutant.copy()
         trivial_mutant.set_token_value(payload)
-        
+
         response = self._uri_opener.send_mutant(trivial_mutant)
 
         # Add data for the persistent xss checking
```

Comment:
We need randomized payloads here to identify source->sink in persistent xss across multiple pages.